pkg/cmd/server/kubernetes/master.go

```diff
@@ -242,6 +243,13 @@ func (c *MasterConfig) RunResourceQuotaManager() {
 	go resourceQuotaManager.Run(c.ControllerManager.ConcurrentResourceQuotaSyncs, utilwait.NeverStop)
 }
 
+func (c *MasterConfig) RunGCController() {
+	if c.ControllerManager.TerminatedPodGCThreshold > 0 {
+		gcController := gccontroller.New(internalclientset.FromUnversionedClient(c.KubeClient), kctrlmgr.ResyncPeriod(c.ControllerManager), c.ControllerManager.TerminatedPodGCThreshold)
```

Comment:
Take a look at something like the persistent volume controllers that make use of service account credentials to run the controller.  That's the model we want to follow for controllers as we move forward.  Limited permissions.